Spicetify Custom App react symbol error

## ℹ Computer information

- Spotify version: Spotify for Windows 1.1.67.586.gbb5ef64e
- Spicetify version: 2.6.1


## 📝 Provide detailed reproduction steps (if any)

1. Try to apply spicetify after upgrading to this Spotify version
2. errors with custom app react symbols

### ✔️ Expected result

_What is the expected result of the above steps?_

Spicetify to apply

### ❌ Actual result

_What is the actual result of the above steps?_

Errors applying spicetify. To note, Spotify doesnt restart and Im still able to use it, but wont apply the config

## 📷 Screenshots

_Are there any useful screenshots?_
![imagen](https://user-images.githubusercontent.com/14024884/131847716-adb08b95-1ec4-463c-aaff-8a55ef6d5a50.png)

Spicetify config state:
![imagen](https://user-images.githubusercontent.com/14024884/131848171-fb62c1e2-ec2a-4599-9b37-f3a7dc046ca7.png)

Just did a fresh install of the new Spotify and spicetify 2.6.2 from AUR. Am able to apply themes just fine, but if I try to apply any of the custom apps I get the same:
```
❯ spicetify apply
spicetify vDev
Overwriting themed assets:
OK
Transferring user.css:
OK
Applying additional modifications:
error Cannot find symbol for Custom app React symbols
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/khanhas/spicetify-cli/src/apply.insertCustomApp.func1({0xc0003e8000, 0x16686f})
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/src/apply/apply.go:180 +0x99b
github.com/khanhas/spicetify-cli/src/utils.ModifyFile({0xc0000a05c0, 0x1e}, 0xc0001239e8)
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/src/utils/utils.go:190 +0x82
github.com/khanhas/spicetify-cli/src/apply.insertCustomApp({0xc0000a05c0, 0x21}, {{0x950058, 0x0, 0x0}, {0xc0000aa200, 0x1, 0x2}, 0x1, 0x1})
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/src/apply/apply.go:155 +0x85
github.com/khanhas/spicetify-cli/src/apply.AdditionalOptions({0xc00001a330, 0x11}, {{0x950058, 0x0, 0x0}, {0xc0000aa200, 0x1, 0x2}, 0x1, 0x1})
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/src/apply/apply.go:34 +0x3df
github.com/khanhas/spicetify-cli/src/cmd.Apply({0x6e13e0, 0x3})
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/src/cmd/apply.go:73 +0x905
main.main()
	/home/nick/.cache/yay/spicetify-cli-git/src/spicetify-cli/spicetify.go:192 +0x4ac
```

If I remove the custom apps from the config, `spicetify apply` runs successfully

--- a/CustomApps/lyrics-plus/Utils.js
+++ b/CustomApps/lyrics-plus/Utils.js
@@ -1,10 +1,11 @@
 class Utils {
     static addQueueListener(callback) {
-        Spicetify.Player.origin2.state.addExtendedStatusListener(callback);
+        Spicetify.Player.origin2.state.addQueueListener(callback);
     }
 
     static removeQueueListener(callback) {
-        Spicetify.Player.origin2.state.removeExtendedStatusListener(callback);
+        Spicetify.Player.origin2.state.queueListeners =
+          Spicetify.Player.origin2.state.queueListeners.filter(v => v == callback);
     }
 
     static convertIntToRGB(colorInt, div = 1) {

--- a/CustomApps/lyrics-plus/Utils.js
+++ b/CustomApps/lyrics-plus/Utils.js
@@ -5,7 +5,7 @@ class Utils {
 
     static removeQueueListener(callback) {
         Spicetify.Player.origin2.state.queueListeners =
-          Spicetify.Player.origin2.state.queueListeners.filter(v => v == callback);
+          Spicetify.Player.origin2.state.queueListeners.filter(v => v != callback);
     }
 
     static convertIntToRGB(colorInt, div = 1) {

--- a/CustomApps/lyrics-plus/index.js
+++ b/CustomApps/lyrics-plus/index.js
@@ -217,7 +217,7 @@ class LyricsContainer extends react.Component {
         this.onQueueChange = async (queue) => {
             this.state.explicitMode = this.state.lockMode;
             this.currentTrackUri = queue.track.uri;
-            const [nextTrack] = queue.future;
+            const nextTrack = queue.next_tracks[0];
             const nextInfo = this.infoFromTrack(nextTrack);
             if (!nextInfo) {
                 this.fetchLyrics(queue.track, this.state.explicitMode);

--- a/src/apply/apply.go
+++ b/src/apply/apply.go
@@ -154,7 +154,7 @@ func getColorCSS(scheme map[string]string) string {
 func insertCustomApp(jsPath string, flags Flag) {
 	utils.ModifyFile(jsPath, func(content string) string {
 		const REACT_REGEX = `lazy\(\(\(\)=>(\w+)\.(\w+)\(\d+\)\.then\(\w+\.bind\(\w+,\d+\)\)\)\)`
-		const REACT_ELEMENT_REGEX = `createElement\(([\w\.]+),\{path:"\/collection"\}`
+		const REACT_ELEMENT_REGEX = `\w+\(\)\.createElement\(([\w\.]+),\{path:"\/collection"\}`
 		reactSymbs := utils.FindSymbol(
 			"Custom app React symbols",
 			content,

--- a/src/apply/apply.go
+++ b/src/apply/apply.go
@@ -153,16 +153,18 @@ func getColorCSS(scheme map[string]string) string {
 
 func insertCustomApp(jsPath string, flags Flag) {
 	utils.ModifyFile(jsPath, func(content string) string {
+		const REACT_REGEX = `lazy\(\(\(\)=>(\w+)\.(\w+)\(\d+\)\.then\(\w+\.bind\(\w+,\d+\)\)\)\)`
+		const REACT_ELEMENT_REGEX = `createElement\(([\w\.]+),\{path:"\/collection"\}`
 		reactSymbs := utils.FindSymbol(
 			"Custom app React symbols",
 			content,
 			[]string{
-				`lazy\(\(function\(\)\{return (\w+)\.(\w+)\(\d+\).then\(\w+\.bind\(\w+,\d+\)\)\}\)\)`})
+				REACT_REGEX})
 		eleSymbs := utils.FindSymbol(
 			"Custom app React Element",
 			content,
 			[]string{
-				`createElement\(([\w\.]+),\{path:"\/collection"\}`})
+				REACT_ELEMENT_REGEX})
 
 		appMap := ""
 		appReactMap := ""
@@ -176,7 +178,7 @@ func insertCustomApp(jsPath string, flags Flag) {
 			appNameArray += fmt.Sprintf(`"%s",`, app)
 
 			appReactMap += fmt.Sprintf(
-				`,spicetifyApp%d=Spicetify.React.lazy((function(){return %s.%s("%s").then(%s.bind(%s,"%s"))}))`,
+				`,spicetifyApp%d=Spicetify.React.lazy((()=>%s.%s("%s").then(%s.bind(%s,"%s"))))`,
 				index, reactSymbs[0], reactSymbs[1],
 				appName, reactSymbs[0], reactSymbs[0], appName)
 
@@ -194,12 +196,12 @@ func insertCustomApp(jsPath string, flags Flag) {
 
 		utils.ReplaceOnce(
 			&content,
-			`lazy\(\(function\(\)\{return [\w\.]+\(\d+\).then\(\w+\.bind\(\w+,\d+\)\)\}\)\)`,
+			REACT_REGEX,
 			`${0}`+appReactMap)
 
 		utils.ReplaceOnce(
 			&content,
-			`\w+\(\)\.createElement\([\w\.]+,\{path:"\/collection"\}`,
+			REACT_ELEMENT_REGEX,
 			appEleMap+`${0}`)
 
 		utils.Replace(

--- a/src/preprocess/preprocess.go
+++ b/src/preprocess/preprocess.go
@@ -263,7 +263,7 @@ func exposeAPIs_main(input string) string {
 	// React Hook
 	utils.ReplaceOnce(
 		&input,
-		`\w+=\(\w+,(\w+)\.lazy\)\(\(function\(\)\{return Promise\.resolve\(\)\.then\(\w+\.bind\(\w+,\w+\)\)\}\)\);`,
+		`\w+=\(\w+,(\w+)\.lazy\)\(\(\(\)=>Promise\.resolve\(\)\.then\(\w+\.bind\(\w+,\w+\)\)\)\);`,
 		`${0}Spicetify.React=${1};`)
 
 	utils.Replace(
@@ -312,14 +312,14 @@ Spicetify.React.useEffect(() => {
 	// React Component: Context Menu and Right Click Menu
 	utils.Replace(
 		&input,
-		`return (\w+\(\)\.createElement\(([\w\.]+),\w+\(\)\(\{\},\w+,\{action:"open",trigger:"right-click"\}\)\))`,
-		`Spicetify.ReactComponent.ContextMenu=${2};Spicetify.ReactComponent.RightClickMenu=${1};return Spicetify.ReactComponent.RightClickMenu`)
+		`=(\w+)=>(\w+\(\)\.createElement\(([\w\.]+),\(\w+,[\w\.]+\)\(\{\},\w+,\{action:"open",trigger:"right-click"\}\)\))`,
+		`=Spicetify.ReactComponent.RightClickMenu=${1}=>${2};Spicetify.ReactComponent.ContextMenu=${3}`)
 
 	// React Component: Context Menu - Menu
 	utils.Replace(
 		&input,
-		`return (\w+\(\)\.createElement\("ul",\w+\(\)\(\{tabIndex:-?\d+,ref:\w+,role:"menu","data-depth":\w+\},\w+\),\w+\))`,//`\w+\(\)\.createElement\([\w\.]+,\{onClose:[\w\.]+,getInitialFocusElement:`,//`\w+\(\)\.createElement\([\w\.]+,\{className:[\w\.]+,onClose:[\w\.]+,onKeyDown:[\w\.]+,onKeyUp:[\w\.]+,getInitialFocusElement:[\w\.]+\},[\w\.]+\)`,
-		`return Spicetify.ReactComponent.Menu=${1}`)
+		`=\(\{children:\w+,onClose:\w+,getInitialFocusElement:\w+\}\)`,
+		`=Spicetify.ReactComponent.Menu${0}`)
 
 	// React Component: Context Menu - Menu Item
 	utils.Replace(

--- a/src/preprocess/preprocess.go
+++ b/src/preprocess/preprocess.go
@@ -271,25 +271,17 @@ func exposeAPIs_main(input string) string {
 		`"data-testid":`,
 		`"":`)
 
-	reAllAPIPromises := regexp.MustCompile(`return (\w+=\w+\.sent),\w+\.next=\d,(Promise.all\(\[(\w\.getSession\(\),)([\w\(\)\.,]+?)\]\))([;,])`)
+	reAllAPIPromises := regexp.MustCompile(`return{version:\w+,(\w+:[\w!\,\(\)\.]+,)+((get\w+:\(\)=>\w+,?)+)}`)
 	allAPIPromises := reAllAPIPromises.FindAllStringSubmatch(input, -1)
 	for _, found := range allAPIPromises {
-		splitted := strings.Split(found[3] + found[4], ",")
-		if len(splitted) > 15 { // Actual number is about 24
-			re := regexp.MustCompile(`\w+\.(\w+)\(\)`)
-			// set t = e.sent, call Promise.all for APIs, then add Spicetify APIs to object
-			code := found[1] + ";" + found[2] + ".then(v => {Spicetify.Platform = {};"
-
-			for apiFuncIndex, apiFunc := range splitted {
-				name := re.ReplaceAllString(apiFunc, `${1}`)
-
-				if strings.HasPrefix(name, "get") {
-					name = strings.Replace(name, "get", "", 1)
-				}
-				code += "Spicetify.Platform[\"" + name + "\"] = v[" + fmt.Sprint(apiFuncIndex) + "];"
+		splitted := strings.Split(found[2], ",")
+		if len(splitted) > 15 { // Actual number is about 34
+			matchMap := regexp.MustCompile(`get(\w+):\(\)=>(\w+),?`)
+			code := "Spicetify.Platform={};"
+			for _, apiFunc := range splitted {
+				matches := matchMap.FindStringSubmatch(apiFunc)
+				code += "Spicetify.Platform[\"" + fmt.Sprint(matches[1]) + "\"]=" + fmt.Sprint(matches[2]) + ";"
 			}
-			code += "});"
-			// Promise.all(...).then(...); return t = e.sent, e.next = 6, Promise.all(...);
 			input = strings.Replace(input, found[0], code + found[0], 1)
 		}
 	}

